libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1807,26 +1802,15 @@ void TypeChecker::requireLValue(Expression const& _expression)
 
 void TypeChecker::typeError(SourceLocation const& _location, string const& _description)
```

Comment:
I also think it is better to move them into the error reporter. Basically there should be one function per kind of error. If you want to restrict the type checker to only use warnings and typeErrors, this can be done in a different way (see my other comment).